File: app/app/src/main/java/me/williamhester/reddit/ui/adapters/VotableAdapter.java
package me.williamhester.reddit.ui.adapters;

import android.view.LayoutInflater;

import java.util.List;

import me.williamhester.reddit.html.HtmlParser;
import me.williamhester.reddit.models.Votable;
import me.williamhester.reddit.ui.ContentClickCallbacks;
import me.williamhester.reddit.ui.VotableCallbacks;

/**
 * Created by william on 6/16/16.
 */
public class VotableAdapter extends ContentAdapter {

  protected List<Votable> votables;

  public VotableAdapter(LayoutInflater inflater,
      HtmlParser htmlParser,
      ContentClickCallbacks contentClickCallbacks,
      VotableCallbacks clickListener,
      List<Votable> votables) {
    super(inflater, htmlParser, contentClickCallbacks, clickListener);
    this.votables = votables;
  }

  @Override
  public int getItemCount() {
    return votables.size();
  }

  @Override
  public Object getItemForPosition(int position) {
    return votables.get(position);
  }
}

File: app/app/src/main/java/me/williamhester/reddit/ui/adapters/VotableAdapter.kt
package me.williamhester.reddit.ui.adapters

import android.view.LayoutInflater

import me.williamhester.reddit.html.HtmlParser
import me.williamhester.reddit.models.Votable
import me.williamhester.reddit.ui.ContentClickCallbacks
import me.williamhester.reddit.ui.VotableCallbacks

/** An adapter that holds [Votable] content. */
class VotableAdapter(
    inflater: LayoutInflater,
    htmlParser: HtmlParser,
    contentClickCallbacks: ContentClickCallbacks,
    clickListener: VotableCallbacks,
    protected var votables: List<Votable>
) : ContentAdapter(inflater, htmlParser, contentClickCallbacks, clickListener) {

  override fun getItemCount(): Int {
    return votables.size
  }

  override fun getItemForPosition(position: Int): Any {
    return votables[position]
  }
}
